applications/Unity.GrantManager/modules/Unity.Flex/src/Unity.Flex.Application/Scoresheets/ScoresheetAppService.cs

```diff
@@ -57,10 +62,15 @@ public virtual async Task<QuestionDto> CreateQuestionInHighestOrderSectionAsync(
 
             lock (_questionLockObject)
             {
+                var questionName = dto.Name.Trim();
+                if(_sectionRepository.HasQuestionWithNameAsync(scoresheetId, questionName).Result)
```

Comment:
Have a look at the worksheets, this is implemented in the Domain model as these are important domain rules (the uniqueness) - you may need some tweaking to the code - but essentially a worksheet or scoresheet is an aggregate of entities that have a set of rules governing the aggregate (you should load, and write at the aggregate level) 

![image](https://github.com/user-attachments/assets/2d319014-6c9e-4329-9c9a-9ec21cc3f89e)